### PR TITLE
Server csp middleware.

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,10 +10,13 @@
   },
   "dependencies": {
     "body-parser": "1.17.1",
+    "compression": "^1.6.2",
     "express": "4.15.2",
     "google-spreadsheet": "2.0.3",
     "guid": "0.0.12",
     "habitat": "3.1.2",
+    "helmet": "^3.5.0",
+    "helmet-csp": "^2.4.0",
```
Comment (cadecairos) [inline package.json]: helmet should come with csp middleware included.

Comment (cadecairos) [reply package.json]: `app.use(helmet.contentSecurityPolicy( ... ))`

File: server.js [hunk 2]
```diff
@@ -3,9 +3,43 @@ require('habitat').load();
...
+var csp = require('helmet-csp');
+var frameguard = helmet.frameguard;
 
 var bodyParser = require('body-parser');
+
+app.set('trust proxy', true);
+
 app.use(bodyParser.json());
+app.use(compression());
+app.use(helmet());
```
Comment (cadecairos) [inline server.js]: lets be more verbose about the security settings we use.

frameguard: https://helmetjs.github.io/docs/frameguard/ - lets set it to 'deny'
hidePoweredBy: https://helmetjs.github.io/docs/hide-powered-by/ just use the default
hsts: https://helmetjs.github.io/docs/hsts/ - max age of one year
ieNoOpen: https://helmetjs.github.io/docs/ienoopen/ - use default
noSniff: https://helmetjs.github.io/docs/dont-sniff-mimetype/ - use default
xssFilter: https://helmetjs.github.io/docs/xss-filter/ - use default

File: server.js [hunk 3]
```diff
@@ -3,9 +3,43 @@ require('habitat').load();
...
+
+app.use(csp({
+  directives:{
+    scriptSrc: ["'self'", "https://*.shpg.org/", "https://www.google-analytics.com/"],
```
Comment (cadecairos) [inline server.js]: add `defaultSrc: [ "'none'" ]`

File: server.js [hunk 4]
```diff
@@ -3,9 +3,43 @@ require('habitat').load();
...
+    connectSrc:["'self'"],
+    childSrc:["'self'"],
+    frameSrc: ["'self'"],
+    imgSrc:["'self'", "https://www.google-analytics.com", "https://*.shpg.org/"],
+    frameAncestors: ["'none'"]
+  }
+}));
+
+app.use(helmet.hsts({
+  maxAge: 90 * 24 * 60 * 60 * 1000 // 90 days
```
Comment (cadecairos) [inline server.js]: as mentioned above, make this one year

**Review by cadecairos — CHANGES_REQUESTED** [review]

**Comment by ScottDowne** [comment]
@cadecairos Updated.

File: server.js [hunk 5]
```diff
@@ -3,9 +3,50 @@ require('habitat').load();
 var express = require('express');
 var app = express();
 var routes = require('./routes');
+var compression = require('compression');
+var helmet = require('helmet');
+var frameguard = helmet.frameguard;
```
Comment (cadecairos) [inline server.js]: small nit - we might as well do what we do elsewhere and call helmet.frameguard instead of referencing it in a local variable.

**Review by cadecairos — APPROVED** [review]
one change, then merge away!

**Pull request merged**